recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "clang": "12",
+            "apple-clang": "12",
+        }
+
+    def configure(self):
+        if self.options.without_python:
+            self.options.rm_safe("python_version")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("tomlplusplus/3.2.0")
+        if not self.options.without_python:
+            self.requires("openssl/1.1.1s")
+            self.requires(f"cpython/{self.options.python_version}")
+            if is_msvc(self):
+                # Required to create import .lib for building extension module.
+                self.options["cpython"].shared = True
+            self.requires("pybind11/2.10.0")
+
+    def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support MacOS at this time"
+            )
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.21.7")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["OPENASSETIO_ENABLE_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+        libcxx = self.settings.get_safe("compiler.libcxx")
+        if libcxx is not None:
+            if libcxx == "libstdc++11":
+                tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = True
+            else:
+                tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = False
+
+        if self.options.without_python:
+            tc.variables["OPENASSETIO_ENABLE_PYTHON"] = False
+        else:
+            tc.variables["OPENASSETIO_ENABLE_PYTHON"] = True
+            tc.variables["Python_EXECUTABLE"] = self._python_exe
+            if is_msvc(self):
+                tc.variables["Python_LIBRARY"] = self._python_windows_lib
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    @property
+    def _python_exe(self):
+        return pathlib.Path(self.deps_user_info["cpython"].python).as_posix()
```

Comment:
```suggestion
        return unix_path(self, os.path.join(self.dependencies["cpython"].cpp_info.bindirs[0], "python"))
```

https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-unix-path

Reply:
Unfortunately:

* Simply joining `python` to `bindirs[0]` is brittle, e.g a debug build will be `python_d.exe` - see `cpython` recipe's `_cpython_interpreter_name` function.
* `unix_path`, when tested locally, just passed through the Windows path unmodified.  It must be a posix path, since the path is baked into CMake config files, and CMake interprets any `\` as an escape sequence. We could escape the `\`s to `\\`, but that's more hassle than just using posix paths.

More on `unix_path`: it also apparently potentially adds a prefix based on the subsystem, which could be important. 

However, I note you didn't flag any of the other paths in the recipe as needing to be wrapped in `unix_path`, e.g. the `_python_windows_lib` method.  Is that an omission?  Or perhaps because `rootpath`/`package_folder` is already appropriately prefixed?  If it's the latter, then we don't have a problem here either, since the `cpython` recipe prefixes `self.package_folder` to the path to the Python binary.

Reply:
I see, the problem is, `deps_user_info` will be deprecated on Conan 2.x and this recipe will be incompatible. However, checking the `cpython` recipe, it's still outdated and not ported to Conan 2.x. Said that, let's keep your way and as soon as we have cpython updated, we can update here too, but using `self.config.get`.

Thank you for your detailed information.

Reply:
```suggestion
        # TODO: update to V2 once cpython is updated
        return pathlib.Path(self.deps_user_info["cpython"].python).as_posix()
```

Reply:
^ committed suggestion (which then auto-resolved this comment thread, so unresolved)

Reply:
^ Comitted suggestion (which resolved this comment thread, so unresolved)